server/utils/passport.ts

```diff
@@ -81,6 +65,26 @@ export class StateStore {
   };
 }
 
+// Every authentication action is routed through the apex domain.
+// But when there is an error, we want to redirect the user on the
+// same domain or subdomain that they originated from (found in state).
```

Comment:
When an auth method catches an error from a provisioner, it should wrap it using this function and re-throw.